src/driver/driver_api.cc

```diff
@@ -225,6 +226,7 @@ std::pair<IRModule, IRModule> SplitDevHostFuncs(IRModule mod_mixed, const Target
       BindTarget(target),
       tir::transform::LowerWarpMemory(),
       tir::transform::Simplify(),
+      tir::transform::LowerCustomDatatypes(),
```

Comment:
Out of curiosity, since the pass is baked in the lowering pass pipeline, does it affect the current perf numbers?

Reply:
Is there a script for gathering the performance data? In the case that there are no custom datatypes registered, the pass shouldn't add much overhead, but I'm certainly happy to validate that by running tests.

Reply:
Yeah, that should be fine though. One quick way to check performance could be running a few models using the master branch against your repo. There should be some scripts in the tutorial, but here is one for running vm and graph runtime:

https://github.com/apache/incubator-tvm/blob/master/tests/python/relay/benchmarking/benchmark_vm.py

Reply:
hi @zhiics thank you for the comment. Here is the data I data gathered using the script you linked, running purely on graph runtime:

|                         | Master          | CustomDatatypes |
|-------------------------|-----------------|-----------------|
| Resnet18 (cpu, gpu)     | 22.19ms, 0.98ms | 21.57ms, 0.98ms |
| Resnet50 (cpu, gpu)     | 49.92ms, 2.86ms | 52.53ms, 2.87ms |
| Mobilenet (cpu, gpu)    | 8.30ms, 0.70ms  | 8.50ms, 0.70ms  |
| Inception_v3 (cpu, gpu) | 70.50ms, 6.18ms | 69.97ms, 6.18ms |

so there is no perf impact on the current numbers

Reply:
Great! Thanks.